### PR TITLE
feat: added support for mysql2 v3.11.5

### PR DESCRIPTION
**Pull request by instanacd**
In version [3.11.5](https://github.com/sidorares/node-mysql2/releases/tag/v3.11.5) of the `mysql2` library, some changes were introduced in the internal structure of the library. Specifically:

- The `query` and `execute` methods were moved to `BaseConnection` in `lib/base/connection.js`.
- The `getConnection`, `query`, and `execute` methods were relocated to `BasePool` in `lib/base/pool.js`.

Updated instrumentation logic to:
- Check if the `Connection` prototype derives from `BaseConnection` for newer versions (v3.11.5+).
- Ensure correct handling of the `Pool` class with changes to the `BasePool` class.

ref INSTA-20850

reference PR https://github.com/sidorares/node-mysql2/pull/3081

### CODE REVIEW

**Comment by aryamohanan** [comment]
Skipped the Couchbase update from 4.4.3 to 4.4.4, as it is addressing in the PR https://github.com/instana/nodejs/pull/1466

**Review by aryamohanan — DISMISSED** [review]
LGTM, waiting for the test to pass

**Review by kirrg001 — CHANGES_REQUESTED** [review]
Could you please add multi testing for "3.11.4" and "^3.11.5"?

File: packages/collector/test/tracing/database/mysql/app.js [hunk 1]
```diff
@@ -14,6 +14,9 @@ process.on('SIGTERM', () => {
 });
 
 const agentPort = process.env.INSTANA_AGENT_PORT;
+if (process.env.MYSQL2_VERSION) {
```
Comment (kirrg001) [inline packages/collector/test/tracing/database/mysql/app.js:17]: issue: MYSQL2_VERSION is set to "null" via the test. We mock it with "mysql2-null".

**Review by kirrg001 — APPROVED** [review]
One issue needs to be fixed. Otherwise pre-approve. LGTM!

**Comment by kirrg001** [comment]
Please use:

> fix(mysql2): handled mysql2 v3.11.5 class structure changes #1467

**feat(mysql2): added support for v3.11.5**

or

**feat: added support for mysql2 v3.11.5**

File: packages/collector/test/tracing/database/mysql/app.js (hunk 1 above)
Comment (aryamohanan) [reply packages/collector/test/tracing/database/mysql/app.js:17]: updated

**Pull request merged**